src/Core/Challenge/Dns/OvhSolver.php

```diff
@@ -0,0 +1,243 @@
+                $topLevelDomain = $this->domainDelegation;
+            }
+
+            $result = $ovh->post('/domain/zone/'.$topLevelDomain.'/record', [
+            'fieldType' => 'TXT',
+            'subDomain' => $subDomain,
+            'target' => $recordValue,
+            'ttl' => 600,
+            ]);
+
+            if (isset($result['message']) && '' !== $result['message']) {
+                $this->logger->error('OVH Exception Post = '.$result['message'].' '.$topLevelDomain);
+                throw new \Exception('OVH Exception Post = '.$result['message'].' '.$topLevelDomain);
+            }
+
+            $this->logger->info('Refresh zone for '.$authorizationChallenge->getDomain());
+            $result = $ovh->post('/domain/zone/'.$topLevelDomain.'/refresh');
+
+            if (isset($result['message']) && '' !== $result['message']) {
+                $this->logger->error('OVH Exception Refresh= '.$result['message'].' '.$topLevelDomain);
+                throw new \Exception('OVH Exception Refresh= '.$result['message'].' '.$topLevelDomain);
+            }
+
+            $this->logger->debug('Create done.');
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function cleanup(AuthorizationChallenge $authorizationChallenge)
+    {
+        return $this->cleanupAll([$authorizationChallenge]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function cleanupAll(array $authorizationChallenges)
+    {
+        Assert::allIsInstanceOf($authorizationChallenges, AuthorizationChallenge::class);
+
+        foreach ($authorizationChallenges as $authorizationChallenge) {
+            $this->logger->debug('Cleanup record id for '.$authorizationChallenge->getDomain());
+
+            $topLevelDomain = $this->getTopLevelDomain($authorizationChallenge->getDomain());
+            $recordName = $this->extractor->getRecordName($authorizationChallenge);
+
+            $subDomain = \str_replace('.'.$topLevelDomain.'.', '', $recordName);
+
+            $client = new Client(['http_errors' => false]);
+
+            $ovh = new Api($this->appKey,
+                $this->appSecret,
+                $this->endPoint,
+                $this->consumerKey,
+                $client);
+
+            if ('' !== $this->domainDelegation) {
+                $topLevelDomain = $this->domainDelegation;
+            }
+
+            $result = $ovh->get('/domain/zone/'.$topLevelDomain.'/record', [
+            'fieldType' => 'TXT', // Filter the value of fieldType property (like) (type: zone.NamedResolutionFieldTypeEnum)
+            'subDomain' => $subDomain, // Filter the value of subDomain property (like) (type: string)
+            ]);
+
+            if (isset($result['message']) && '' !== $result['message']) {
+                $this->logger->error('OVH Exception Get= '.$result['message'].' '.$topLevelDomain);
+                throw new \Exception('OVH Exception Get= '.$result['message'].' '.$topLevelDomain);
```

Comment:
You can adapt the message I proposed in previous comments here and in the other exceptions